.github/workflows/wheels.yml

```
@@ -26,6 +26,9 @@ on:
     branches:
       - main
       - maintenance/**
+  push:
+    branches:
+      - maintenance/**
```

```suggestion
    branches:
      - maintenance/**
    tags:
      - v*
```

Please check https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushbranchestagsbranches-ignoretags-ignore, only defining `branches` will not triggered by the tag

---

But the tag is typically pushed to a maintenance branch, so is it necessary to have both?

---

There's then a check to see if the tag starts with `v`.

---

> If you define only tags/tags-ignore or only branches/branches-ignore, the workflow won't run for events affecting the undefined Git ref.

I am sure `refs/tags` will not triggered by `branches` as I met this issue before in my project.

---

Agreed, it should just be `tags:`. Triggering on every push to `maintenance` seems wrong, and will include all regular/merge commits.

---

`tags`-only matches the [personal project](https://github.com/refnx/refnx/blob/c5282834a5055348af57755db6dda0ddef436d02/.github/workflows/release.yml#L6) you linked to @andyfaff. So I think the suggestion right above should do it. If you agree, let's merge and backport that, so @charris can try it.